### PR TITLE
Added two new pipes: Fuel and Auxiliary Pipes, and added a new Airlock mode to cycle external air.

### PR DESCRIPTION
**Pull request by Generalcamo**
This PR adds two new types of pipes: Fuel, and Auxiliary piping. These are intended to allow for more compact setups on shuttles where space can be a premium, but are useful in many situations. 

During the course of this PR, I realized that the airlocks in Aurora make no distinction between cycling exterior and interior air. This could be problematic if, for example, the Intrepid lands on a planet with a phoron atmosphere, and the airlock cycle sucks in phoron into the clean air. This PR adds a new mode to handle this which can cycle exterior air externally, but maintains backwards compatibility.

This PR demonstrates these functions in an updated Spark and Intrepid.

Due to an issue with the new underlays, plus a prior discussion, Keekenox#1337's Vents and Scrubbers were ported to Aurora (these are used in Polaris, Bay, Eris, among others).

### Asset Licenses
The following assets that **have not** been created by myself are included in this PR:

| Path | Original Author | License |
| --- | --- | --- |
| icons/atmos/vent_scrubber.dmi | Keekenox#1337 (For Baystation12) | Public Domain |
| icons/atmos/vent_pump.dmi | Keekenox#1337 (For Baystation12) | Public Domain |
| icons/atmos/pipes.dmi | Arokha (Polaris) | GNU Affero General Public License 3.0 | 
| icons/atmos/pipe_underlays.dmi | Arokha (Polaris) | GNU Affero General Public License 3.0 | 
| icons/atmos/pump.dmi | Arokha (Polaris) | GNU Affero General Public License 3.0 |

### CODE REVIEW

**Comment by Generalcamo** [comment]
!review

**Review by alsoandanswer — APPROVED** [review]

File: code/game/machinery/pipe/construction.dm [hunk 1]
```diff
@@ -611,6 +785,52 @@
...
+			var/obj/machinery/atmospherics/pipe/manifold/hidden/fuel/M = new( src.loc )
+			M.color = color
+			M.set_dir(dir)
+			M.initialize_directions = pipe_dir
+			//M.New()
+			var/turf/T = M.loc
+			M.level = !T.is_plating() ? 2 : 1
+			M.atmos_init()
+			if (!M)
+				to_chat(usr, "There's nothing to connect this manifold to! (with how the pipe code works, at least one end needs to be connected to something, otherwise the game deletes the segment)")
```
Comment (SleepyGemmy) [inline code/game/machinery/pipe/construction.dm:798]:
```suggestion
				to_chat(usr, "There's nothing to connect this manifold to!")
```

File: code/game/machinery/pipe/construction.dm [hunk 2]
```diff
@@ -611,6 +785,52 @@
...
+			var/obj/machinery/atmospherics/pipe/manifold/hidden/aux/M = new( src.loc )
+			M.color = color
+			M.set_dir(dir)
+			M.initialize_directions = pipe_dir
+			//M.New()
+			var/turf/T = M.loc
+			M.level = !T.is_plating() ? 2 : 1
+			M.atmos_init()
+			if (!M)
+				to_chat(usr, "There's nothing to connect this manifold to! (with how the pipe code works, at least one end needs to be connected to something, otherwise the game deletes the segment)")
```
Comment (SleepyGemmy) [inline code/game/machinery/pipe/construction.dm:821]:
```suggestion
				to_chat(usr, "There's nothing to connect this manifold to!")
```

File: code/game/machinery/pipe/construction.dm [hunk 3]
```diff
@@ -722,6 +942,20 @@
...
+			if (pipename)
+				V.name = pipename
+			var/turf/T = V.loc
+			V.level = !T.is_plating() ? 2 : 1
+			V.atmos_init()
+			V.build_network()
+			if (V.node)
+				V.node.atmos_init()
+				V.node.build_network()
+
```
Comment (SleepyGemmy) [inline code/game/machinery/pipe/construction.dm:958]:
```suggestion
```

File: icons/atmos/vent_pump.dmi
Comment (SleepyGemmy) [inline icons/atmos/vent_pump.dmi:1]: `in_expanded` and `out_expanded` doesn't seem used. please remove them if this is the case.

File: icons/atmos/vent_scrubber.dmi
Comment (SleepyGemmy) [inline icons/atmos/vent_scrubber.dmi:1]: `in_expanded` and `on_expanded` doesn't seem used. please remove them if this is the case.

File: code/game/machinery/pipe/construction.dm (hunk 1 above)
Comment (Generalcamo) [reply code/game/machinery/pipe/construction.dm:798]: This is present in -all- the manifolds. I object to changing only these two pipes.

File: code/game/machinery/pipe/construction.dm (hunk 2 above)
Comment (Generalcamo) [reply code/game/machinery/pipe/construction.dm:821]: Ditto, see above

File: icons/atmos/vent_pump.dmi
Comment (SleepyGemmy) [inline icons/atmos/vent_pump.dmi:1]: image
map icon state sprite misaligned (see third dir state).

**Review by SleepyGemmy — APPROVED** [review]
code is fine.

**Review by SleepyGemmy — APPROVED** [review]

**Pull request merged**